utils/publish-canary.js

```diff
@@ -11,28 +11,72 @@ const {
   TRAVIS_BUILD_URL = 'https://travis-ci.org/Workday/canvas-kit/branches',
 } = process.env;
 
+const isPrerelease = TRAVIS_BRANCH.match(/^prerelease\/v\d*$/g);
 const data = {};
-let preid;
+let distTag;
 
 if (TRAVIS_BRANCH === 'master') {
-  preid = 'next';
-} else if (TRAVIS_BRANCH.match(/^prerelease\/v\d*$/g)) {
-  preid = 'prerelease';
+  distTag = 'next';
+} else if (isPrerelease) {
+  // distTag = 'prerelease-next';
+  console.error('Prerelease canary builds disabled.');
+  process.exit(0);
 } else {
   console.error('No travis branch provided');
   process.exit(1);
 }
 
-cmd('git rev-parse --short HEAD')
+const slackAnnouncement = attachment => {
+  request.post(
+    SLACK_WEBHOOK,
+    {
+      json: {
+        attachments: [
+          {
+            ...attachment,
+            author_link: TRAVIS_BUILD_URL,
+            ts: Date.now(),
+          },
+        ],
+      },
+    },
+    (error, response, body) => {
+      if (error) {
+        throw error;
+      }
+    }
+  );
+};
+
+cmd('git diff --name-only HEAD HEAD^')
+  .then(filesChangedInMerge => {
+    if (filesChangedInMerge.includes('CHANGELOG.md')) {
```

Comment:
What if we change/fix the `CHANGELOG.md` file as part of previous commit? Is there a better way to determine if the last merge was a commit?

Reply:
Not quite sure what you mean here. Are you talking about the case where we have a merge commit that updates the changelog that is not a release?

If so, I discussed that case with @NicholasBoll and we agreed it was probably fine if we cause a canary build to be skipped every once in a while. It seemed like a better option than the alternative of looking for `chore: Release` in the commit message (less room for user error)